presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionFactory.java

```diff
@@ -157,6 +157,9 @@ public Session createSession(
             SessionPropertyManager sessionPropertyManager)
     {
         accessControl.checkCanSetUser(identity.getPrincipal().orElse(null), identity.getUser());
+        if (catalog != null) {
```

Comment:
I thought that we should have a check about whether a user can access the catalog before calling `setCatalog()` method. But I didn't see think of this case when a query may not be accessing the catalog. I have removed this code for now.